### PR TITLE
Add map valued (string->string, string->int) flags.

### PR DESCRIPTION
**Pull request by tamalsaha**
Format: --myflag=a=1,b=2

ref: #129

### CODE REVIEW

**Comment by tamalsaha** [comment]
@eparis, mind taking a look?

**Comment by sgraham785** [comment]
@tamalsaha why was this closed?  it's still a useful implementation unless there is now some other way to accomplish now?

**Comment by eparis** [comment]
somehow I totally missed this....   sorry....

File: string_to_string.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+func newStringToStringValue(val map[string]string, p *map[string]string) *stringToStringValue {
+	ssv := new(stringToStringValue)
+	ssv.value = p
+	*ssv.value = val
+	return ssv
+}
+
+// Format: a=1,b=2
+func (s *stringToStringValue) Set(val string) error {
+	ss := strings.Split(val, ",")
```
Comment (eparis) [inline string_to_string.go]: Should this actually be a CSV decode, instead of just splitting on ,?  What is there is a comma in the value?

**Comment by tamalsaha** [comment]
Thanks @eparis ! I have addressed your comment. PTAL.

**Comment by eparis** [comment]
that's really awesome! If you'll write a test of the csv stuff, I'll merge it!

**Comment by tamalsaha** [comment]
> If you'll write a test of the csv stuff, I'll merge it!

I added some tests here:

- https://github.com/spf13/pflag/pull/133/files#diff-5f559d5cd1fe9b2d957bdc7eee7c427dR63
- https://github.com/spf13/pflag/pull/133/files#diff-5f559d5cd1fe9b2d957bdc7eee7c427dR116

**Comment by tamalsaha** [comment]
Added even more tests, Search for `5,6`, `7,8` . PTAL.

**Comment by yashbhutwala** [comment]
@tamalsaha Does `StringToStringVar` support --myflag=a=1,b=2,3

**Comment by tamalsaha** [comment]
It is not supported.See the comment https://github.com/spf13/pflag/pull/179

**Comment by yashbhutwala** [comment]
I see, OK.  This [comment](https://github.com/spf13/pflag/pull/179#issue-208765984) indicates that `--f3 "c=5,6",d=7` is possible, however, I haven't been able to use it.

**Comment by tamalsaha** [comment]
Try passing this flag like this `--myflag a=1,"b=2,3"` . It should work.

**Comment by yashbhutwala** [comment]
Nope, getting an error: `flag:  value2 must be formatted as key=value`

**Comment by tamalsaha** [comment]
Here is the code that does the parsing. https://github.com/spf13/pflag/pull/179/files#diff-1ace98db8b8888abf672baa63775bc05R24

This should work: https://play.golang.org/p/rIAPpDsiHJQ

**Comment by yashbhutwala** [comment]
OK @tamalsaha this worked for me: `--f3 'a=1,"b=5,6"'`.  So instead of `` wrapping the whole parameter, it's ''

**Pull request merged**